src/x/xSum.ts

```diff
@@ -27,9 +27,9 @@ export function xSum(
   const { fromIndex = 0, toIndex = array.length - 1 } = options;
   xCheck(array);
 
-  let sumValue = array[fromIndex];
+  let sumValue = array[fromIndex] as number;
   for (let i = fromIndex + 1; i <= toIndex; i++) {
-    sumValue += array[i];
+    sumValue += array[i] as number;
```

Comment:
The fact that you need to put as number is a problem. The reason is that we allow in NumberArray a bigint64 typed array and bigints are treated differently in javascript.
I have removed the BigInt from NumberArray in cheminfo-types. Please update the library and remove the 'as number' that you added in many places in the code.